sphinx/environment/__init__.py

```diff
@@ -193,6 +198,10 @@ def __init__(self, app):
         # attributes of "any" cross references
         self.ref_context = {}       # type: Dict[unicode, Any]
 
+        # set up environment
+        if app:
+            self.setup(app)
```

Comment:
I made `app` argument optional, but I don't add DeprecationWarning here. Because I don't have strong opinion for it. Either is fine to me.